atc/exec/retry_error_step.go

```diff
@@ -40,26 +42,30 @@ func (step RetryErrorStep) Run(ctx context.Context, state RunState) error {
 	logger := lagerctx.FromContext(ctx)
 	runErr := step.Step.Run(ctx, state)
 	if runErr != nil && step.toRetry(logger, runErr) {
-		logger.Info("retriable", lager.Data{"error": runErr.Error()})
+		logger.Info("retryable", lager.Data{"error": runErr.Error()})
 		step.delegate.Errored(logger, fmt.Sprintf("%s, will retry ...", runErr.Error()))
-		runErr = Retriable{runErr}
+		runErr = Retryable{runErr}
 	}
 	return runErr
 }
 
 func (step RetryErrorStep) toRetry(logger lager.Logger, err error) bool {
-	switch err.(type) {
-	case transport.WorkerMissingError, transport.WorkerUnreachableError:
+	var transportErr *transport.WorkerMissingError
+	var unreachable *transport.WorkerUnreachableError
+	re := regexp.MustCompile(`worker .+ disappeared`)
+	if ok := errors.Is(err, transportErr) || errors.Is(err, unreachable); ok {
```

Comment:
https://blog.golang.org/go1.13-errors

> In the simplest case, the errors.Is function behaves like a comparison to a sentinel error, and the errors.As function behaves like a type assertion. When operating on wrapped errors, however, these functions consider all the errors in a chain.

I think you want to be using `errors.As` here, not `errors.Is`